src/auspex/experiment.py

```diff
@@ -308,14 +310,30 @@ def update_descriptors(self):
 
         done = True
         while True:
-            await self.sweeper.update()
+            # Increment the sweeper and get the current set of values
+            sweep_values = await self.sweeper.update()
```

Comment:
what does `self.sweeper.udpate()` return? a list of tuples? if so, a comment would be useful here.